File: vuln-sandbox/python/taint/sqli-1hop.py
# Taint: SQL Injection — 1-hop (Python)
# Source: request.args.get('id') → Sink: cursor.execute
# Expected: TruePositive (taint/cwe89)
from flask import Flask, request
import sqlite3

app = Flask(__name__)

@app.route('/user')
def get_user():
    user_id = request.args.get('id')           # SOURCE: HTTP request param
    conn = sqlite3.connect('db.sqlite')
    cursor = conn.cursor()
    cursor.execute(f"SELECT * FROM users WHERE id = {user_id}")  # SINK
    return str(cursor.fetchall())

File: vuln-sandbox/node/taint/sqli-1hop.js
// Taint: SQL Injection — 1-hop (JS)
// Source: req.query.id → Sink: db.query (same function)
// Expected: TruePositive (taint/cwe89)
const express = require('express');
const app = express();

app.get('/user', async (req, res) => {
    const id = req.query.id;                    // SOURCE: HTTP request param
    const result = await db.query(`SELECT * FROM users WHERE id = ${id}`); // SINK
    res.json(result);
});
